addons/web/static/tests/utils.js

```diff
@@ -644,3 +644,82 @@ class Contains {
 export async function contains(selector, options) {
     await new Contains(selector, options).run();
 }
+
+const stepState = {
+    expectedSteps: null,
+    deferred: null,
+    timeout: null,
+    steps: [],
+
+    clear() {
+        clearTimeout(stepState.timeout);
+        this.timeout = null;
+        this.deferred = null;
+        this.steps = [];
+        this.expectedSteps = null;
+    },
+
+    check() {
+        this.timeout = setTimeout(() => {
+            QUnit.config.current.assert.verifySteps(this.expectedSteps);
+            this.deferred.reject(
+                new Error(`Timeout while waiting for steps: "${this.expectedSteps}"`)
+            );
+            this.clear();
+        }, 2000);
+        if (
+            this.expectedSteps.length !== this.steps.length ||
+            !this.expectedSteps.every((s, i) => s === this.steps[i])
+        ) {
+            return;
+        }
+        QUnit.config.current.assert.verifySteps(this.expectedSteps);
+        this.deferred.resolve();
+        this.clear();
+    },
+};
+
+if (window.QUnit) {
+    QUnit.testDone(() => {
+        if (stepState.expectedSteps) {
+            QUnit.config.current.assert.verifySteps(stepState.expectedSteps);
+        }
+        stepState.clear();
```

Comment:
I would also have call `check` with `crashOnFail` to avoid duplication and inconstencies. Same as `runOnce` function of contains.